pkg/services/ngalert/notifier/external_alertmanager.go

```diff
@@ -66,11 +82,87 @@ func newExternalAlertmanager(cfg externalAlertmanagerConfig, orgID int64) (*exte
 	}, nil
 }
 
+func (am *externalAlertmanager) ApplyConfig(ctx context.Context, config *models.AlertConfiguration) error {
+	cfg, err := Load([]byte(config.AlertmanagerConfiguration))
+	if err != nil {
+		return err
+	}
+
+	return am.postConfig(ctx, cfg)
+}
+
 func (am *externalAlertmanager) SaveAndApplyConfig(ctx context.Context, cfg *apimodels.PostableUserConfig) error {
-	return nil
+	if err := am.postConfig(ctx, cfg); err != nil {
+		return err
+	}
+	return am.saveConfig(ctx, cfg, false)
 }
 
 func (am *externalAlertmanager) SaveAndApplyDefaultConfig(ctx context.Context) error {
+	cfg, err := Load([]byte(am.defaultConfig))
+	if err != nil {
+		return err
+	}
+
+	if err := am.postConfig(ctx, cfg); err != nil {
+		return err
+	}
+
+	return am.saveConfig(ctx, cfg, true)
+}
+
+func (am *externalAlertmanager) saveConfig(ctx context.Context, cfg *apimodels.PostableUserConfig, isDefault bool) error {
+	b, err := json.Marshal(&cfg)
+	if err != nil {
+		return fmt.Errorf("failed to serialize to the Alertmanager configuration: %w", err)
+	}
+
+	cmd := ngmodels.SaveAlertmanagerConfigurationCmd{
+		AlertmanagerConfiguration: string(b),
+		ConfigurationVersion:      fmt.Sprintf("v%d", ngmodels.AlertConfigurationVersion),
+		OrgID:                     am.orgID,
+		LastApplied:               time.Now().UTC().Unix(),
+		Default:                   isDefault,
+	}
+	return am.configStore.SaveAlertmanagerConfiguration(ctx, &cmd)
+}
+
+func (am *externalAlertmanager) postConfig(ctx context.Context, cfg *apimodels.PostableUserConfig) error {
+	b, err := yaml.Marshal(cfg)
```

Comment:
Do I understand correctly, that here `cfg` is supposed to be decrypted before sending to the external manager?